app/javascript/react/containers/ReviewFormContainer.js

```diff
@@ -0,0 +1,183 @@
+import React from 'react'
+import ReviewFormInput from '../components/ReviewFormInput'
+
+class ReviewFormContainer extends React.Component {
+  constructor(props){
+    super(props)
+    this.state = {
+      overall_rating: null,
+      body: '',
+      food_rating: null,
+      atmosphere_rating: null,
+      shows_rating: null,
+      staff_rating: null,
+      price_rating: null,
+      rides_rating: null,
+      errors: {}
+    }
+    this.handleChange = this.handleChange.bind(this)
+    this.handleClear = this.handleClear.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.validateOverallRatingEntry = this.validateOverallRatingEntry.bind(this)
+  }
+
+  handleChange(event){
+    if (event.target.name == "overall_rating") {
+      this.validateOverallRatingEntry(event.target.value)
+    }
+    this.setState({
+      [event.target.name]: event.target.value
+    })
+  }
+
+  handleSubmit(event){
+    event.preventDefault()
+
+    if (this.validateOverallRatingEntry(this.state.overall_rating)){
+      let newReview = {
+        overall_rating: this.state.overall_rating,
+        amusement_park_id: this.props.id
+      }
+
+      if (this.state.food_rating != null) {
```

Comment:
these conditionals are unnecessary. You can just assign these directly. If they're nil JS will do the right thing.